ironfish/src/network/peers/peer.ts

```diff
@@ -259,36 +228,40 @@ export class Peer {
 
   /**
    * Sets a WebRTC connection on the peer, moving it into the CONNECTING state if necessary.
-   * Ignores the connection if the peer already has a WebRTC connection.
+   * Returns the existing connection if the peer already has a WebRTC connection.
    * @param connection The WebRTC connection to set
    */
-  setWebRtcConnection(connection: WebRtcConnection): void {
+  setWebRtcConnection(connection: WebRtcConnection): WebRtcConnection | null {
+    let existingConnection = null
     if (this.state.type !== 'DISCONNECTED' && this.state.connections.webRtc) {
-      this.logger.warn('Already have a WebRTC connection, ignoring the new one')
-      return
+      existingConnection = this.state.connections.webRtc
```

Comment:
Is there a reason we've changed this to not early return anymore? Don't we still want to keep the existing connection if we have one?

Reply:
We added a new function to differentiate between replacing an existing connection and setting a connection if it does not exist